mdata/cache/if.go

```diff
@@ -6,6 +6,7 @@ import (
 
 type Cache interface {
 	Add(string, uint32, chunk.IterGen)
+	CacheIfHot(string, uint32, *chunk.IterGen)
```

Comment:
it's ok that the type is larger than the pointer. the problem with pointers is more about the dereferencing overhead and cache misses causing stalls (I don't have evidence to back this up, more like a gut feeling)